documents-storage-jcr/src/main/java/org/exoplatform/documents/storage/jcr/JCRDocumentFileStorage.java

```diff
@@ -320,7 +320,7 @@ public List<AbstractNode> getFolderChildNodes(DocumentFolderFilter filter,
           String statement = getFolderDocumentsQuery(parent.getPath(), sortField, sortDirection);
           Query jcrQuery = session.getWorkspace().getQueryManager().createQuery(statement, Query.SQL);
           ((QueryImpl)jcrQuery).setOffset(offset);
```

Comment:
shouldn't we update the offset with the index of the last returned element ? 
What if limit*2 elements got filtered and we get less than limit elements ?